tags.go

```diff
@@ -0,0 +1,42 @@
+package distribution
+
+// Tag is a named reference to a repository object. Target generally refers to
+// a manifest.
+type Tag interface {
+	Blob // io access to serialized content.
+
+	// Name returns the fully qualified name of the object. For example, if
+	// the object is in repository "stevvooe/test" and has tag "latest", the
+	// name will be "stevvooe/test:latest".
+	Name() string
+
+	// Description returns a human-readable description of the tag.
+	Description() string
+
+	// Target identifies the blob targeted by this tag.
+	Target() Descriptor
+}
+
+// TagService provides access to information about tagged objects.
+type TagService interface {
+	All() (Tag, error)
+
+	// Get retrieves the digest for the provided tag.
+	Get(name string) (Tag, error)
```

Comment:
The tag service resolves from a name to a digest. Going the other way, one would digest a tag object. If one wants access to tag content as a blob, using the blob service is the right approach. I am not sure of a use case in which one would fetch a tag by its digest.